### PR TITLE
add inline ads feature

### PR DESCRIPTION
**Pull request by mehdignu**
Fixes #
(comma separated list of tickets/issues/bugs fixed by the PR)

## Proposed Changes

For the ad integration on articles we need a dynamic functionality. The ad integration should be depending on paragraphs and the numbers of characters.

The following rules should be respected:

- an adslots should be integrated after every paragraph (with 200 characters)
- an paragraph should have at least 200 characters
- an adslot should only be integrated if another paragraph follows
- no integration before and after one single paragraph
- a item of a list could be a paragraph if it has more than 200 characters
- no integration before a list
- no integration after headlines
- the number of characters should be adjustable in den backend
- has a paragraph less than 200 characters the next paragraph(s) should be added till the numbers of 200 characters is reached

License: AGPLv3

### CODE REVIEW

File: src/SWP/Bundle/CoreBundle/Twig/ArticleBodyExtension.php [hunk 1]
```diff
@@ -49,9 +49,70 @@ public function getFunctions(): array
...
 
+    public function setInlineAds(string $body): array
+    {
+        # split the body into paragraphs
+        $paragraphs = explode("</p>", $body);
+
+        $tmp = [];
+        $splittedBody = [];
+        $parLength = 0;
+        $paragraphChars = $_ENV['ADS_CHARACTERS'];
```
Comment (ahilles107) [inline src/SWP/Bundle/CoreBundle/Twig/ArticleBodyExtension.php:64]: You can't do this like that. Publisher (and symfony) have other way for accessing env variables. It should be passed to service via DI.

**Comment by ahilles107** [comment]
where we can find tests?

File: .env [hunk 2]
```diff
@@ -124,3 +124,6 @@ MAILER_DSN=smtp://localhost
 ###< symfony/mailer ###
 
 FROM_EMAIL=contact@publisher.test
+
+# after how many charachters in the body's paragraph the ad should to be placed
```
Comment (takeit) [inline .env:128]: there is a typo in `charachters`

File: src/SWP/Bundle/CoreBundle/Tests/Twig/InlineAdsTest.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+use SWP\Bundle\FixturesBundle\WebTestCase;
+
+class InlineAdsTest extends WebTestCase
+{
+    /**
+     * @var \Twig_Environment
+     */
+    private $twig;
+
+    public function setUp()
```
Comment (takeit) [inline src/SWP/Bundle/CoreBundle/Tests/Twig/InlineAdsTest.php:26]: please add `:void` return type to all the functions in this class and `declare(strict_types=1);` after `<?php`

**Review by takeit — CHANGES_REQUESTED** [review]
@mehdignu please see my comments and we are ready to merge. The PR LGTM.

**Pull request closed without merging**